llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp

```diff
@@ -2550,6 +2550,15 @@ bool IRTranslator::translateKnownIntrinsic(const CallInst &CI, Intrinsic::ID ID,
     MIRBuilder.buildInstr(TargetOpcode::G_RESET_FPMODE, {}, {});
     return true;
   }
+  case Intrinsic::vscale: {
+    LLT DstTy = getLLTForType(*CI.getType(), *DL);
+    auto IntN = IntegerType::get(MF->getFunction().getContext(),
+                                 DstTy.getScalarSizeInBits());
+    ConstantInt *CInt = ConstantInt::get(IntN, 1);
+    MIRBuilder.buildInstr(TargetOpcode::G_VSCALE, {getOrCreateVReg(CI)}, {})
+        .addCImm(CInt);
```

Comment:
Move this into a buildVScale in the MIRBuilder?

Reply:
Done.